cmd/timescale-prometheus/main.go

```diff
@@ -444,6 +458,26 @@ func checkDependencies(cfg *pgclient.Config, appVersion pgmodel.VersionInfo) err
 	return pgmodel.CheckDependencies(db, appVersion)
 }
 
+// isDBVersionSafe verifies if the compatible version of the database.
+func isDBVersionSafe(cfg *pgclient.Config) (bool, error) {
```

Comment:
This should go into pkg/pgmodel/extension.go and should be like checkExtensionVersion (call new func `checkTimescaleVersion`). We should have 3 buckets Everything `>=1.7.3 && <2.0.0` is fine. `>=1.7.0 && <1.7.3` is a warning `<1.7.0 || >=2.0.0` is an error.

Reply:
We should also refactor `func fetchInstalledExtensionVersion` in extensions.go to take the extension name so we could reuse that code.

Reply:
Done and used the refactored `func fetchInstalledExtensionVersion` way.